CharlieBackend.Business/Services/LessonService.cs

```diff
@@ -22,7 +23,7 @@ public LessonService(IUnitOfWork unitOfWork, IMapper mapper)
             _mapper = mapper;
         }
 
-        public async Task<LessonDto> CreateLessonAsync(CreateLessonDto lessonDto)
+        public async Task<Result<LessonDto>> CreateLessonAsync(CreateLessonDto lessonDto)
```

Comment:
Don't forget to call .ToActionResult in controller action. Otherwise, it will change schema of response body.